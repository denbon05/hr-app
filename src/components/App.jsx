import React from 'react';
import AddModal from './AddModal';
import WorkersSheet from './WorkersSheet';
import NewTaskForm from './NewTaskForm.jsx';
import Tasks from './Tasks.jsx';
import NavBar from './NavBar.jsx';

const App = () => (
  <>
    <NavBar />
    <AddModal />
    <WorkersSheet />
    {/* <NewTaskForm />
    <Tasks /> */}
  </>
);
export default App;