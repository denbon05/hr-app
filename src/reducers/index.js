import _ from 'lodash';
import { combineReducers } from 'redux';
import { handleActions } from 'redux-actions';
// import { reducer as formReducer } from 'redux-form';
import * as actions from '../actions/index.js';

// MODALS

const addModalsState = {
  allIds: [],
  byId: {},
  modallAddOn: false,
  submitOn: false,
  name: null,
  phone: null,
  address: null,
  passportNumb: null,
  birthday: null,
  job: null,
  polandFirm: false,
  bankAccountNum: null,
  wayOfStay: null,
  startVisa: null,
  endVisa: null,
  arriveDate: null,
  departureDate: null,
  countLegalDayLost: null,
  datePIP: null,
  kartaPobytuDate: null,
  notes: null,
};

const addModal = handleActions({
  [actions.toogleAddModal](state) {
    const { modallAddOn } = state;
    state.modallAddOn = !modallAddOn;

    return { ...state };
  },
  [actions.onchangeName](state, { payload : { value } }) {
    state.name = value;
    return { ...state };
  },
  [actions.onchangePhoneNumber](state, { payload: { value } }) {
    state.phone = value;
    return { ...state };
  },
  [actions.onchangeAddress](state, { payload: { value } }) {
    state.address = value;
    return { ...state };
  },
  [actions.onchangePassportNumber](state, { payload: { value } }) {
    state.passportNumb = value;
    return { ...state };
  },
  [actions.selctBirthDayDate](state, { payload: { value } }) {
    state.birthday = value;
    return { ...state };
  },
  [actions.onchangeJob](state, { payload: { value } }) {
    state.job = value;
    return { ...state };
  },
  [actions.polandFirmToogle](state) {
    const { polandFirm } = state;
    state.polandFirm = !polandFirm;
    return { ...state };
  },
  [actions.onchangeBankAccountNum](state, { payload: { value } }) {
    state.bankAccountNum = value;
    return { ...state };
  },
  [actions.chooseWayOfStay](state, { payload: { id } }) {
    state.wayOfStay = id;
    return { ...state };
  },
  [actions.getStartVisaDate](state, { payload: { value } }) {
    state.startVisa = value;
    return { ...state };
  },
  [actions.getEndVisaDate](state, { payload: { value } }) {
    state.endVisa = value;
    return { ...state };
  },
  [actions.selctArriveDate](state, { payload: { value } }) {
    state.arriveDate = value;
    return { ...state };
  },
  [actions.selectDepartureDate](state, { payload: { value } }) {
    state.departureDate = value;
    return { ...state };
  },
  [actions.onchangeCountLegalDays](state, { payload: { value } }) {
    state.countLegalDayLost = value;
    return { ...state };
  },
  [actions.selectPIPDate](state, { payload: { value } }) {
    state.datePIP = value;
    return { ...state };
  },
  [actions.selectConclusionDate](state, { payload: { value } }) {
    state.kartaPobytuDate = value;
    return { ...state };
  },
  [actions.onchangeNotes](state, { payload: { value } }) {
    state.notes = value;
    return { ...state };
  },
  [actions.addWorker](state, { payload: { id } }) {
    const { byId, allIds, modallAddOn } = state;
    allIds.push(id);
    const newWorker = Object.entries(state).reduce((acc, [key, value]) => {
      if (
        key === 'modallAddOn'
        || key === 'submitOn'
        || key === 'allIds'
        || key === 'byId'
      ) {
        return acc;
      }
  
      acc[key] = value;
      state[key] = key === 'polandFirm' ? false : null;
      return acc;
    }, {});
    byId[id] = { ...newWorker, state: 'working' };
    state.modallAddOn = !modallAddOn;
    return { ...state };
  },
}, addModalsState);

// CONFIGURATION

const configState = {};

const configuration = handleActions({
  [actions.setConfiguration](state) {

  }
}, configState);

// REMINDER

const reminder = handleActions({
  [actions.clickBell](state) {
    const { bellClicked } = state;
    state.bellClicked = !bellClicked;
    return { ...state };
  }
}, { bellClicked: false });

// const tasks = handleActions({
//   [actions.confirmAdding](state, { payload: { inputedWorkerInfo } }) {
//     const { byId, allIds } = state;
//     return {
//       byId: { ...byId, [task.id]: task },
//       allIds: [task.id, ...allIds],
//     };
//   },
//   [actions.removeTask](state, { payload: { id } }) {
//     const { byId, allIds } = state;
//     return {
//       byId: _.omit(byId, id),
//       allIds: _.without(allIds, id),
//     };
//   },
//   [actions.toggleTaskState](state, { payload: { id } }) {
//     const task = state.byId[id];
//     const newState = task.state === 'active' ? 'finished' : 'active';
//     const updatedTask = { ...task, state: newState };
//     return {
//       ...state,
//       byId: { ...state.byId, [task.id]: updatedTask },
//     };
//   },
// }, { byId: {}, allIds: [], openAddModal: false,  });

// SEARCH

const navForm = handleActions({
  [actions.clickSearch](state) {

  },
}, { currentValue: '', searchOn: false });

export default combineReducers({
  addModal,
  configuration,
  reminder,
  navForm,
  // tasks,
  // form: formReducer,
});