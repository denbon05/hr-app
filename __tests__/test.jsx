// @ts-check

import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import reducers from '../src/reducers/index.js';
import App from '../src/components/App.jsx';

Enzyme.configure({ adapter: new Adapter() });

test('Store', () => {
  const store = createStore(reducers);

  const vdom = (
    <Provider store={store}>
      <App />
    </Provider>
  );
  const wrapper = mount(vdom);
  expect(wrapper.render()).toMatchSnapshot();

  const newTaskInput = wrapper.find('input[type="text"]');
  const newTaskSubmit = wrapper.find('input[type="submit"]');

  newTaskInput.simulate('change', { target: { value: 'na-na' } });
  expect(wrapper.render()).toMatchSnapshot();

  newTaskSubmit.simulate('submit');
  expect(wrapper.render()).toMatchSnapshot();

  newTaskInput.simulate('change', { target: { value: 'another task' } });
  expect(wrapper.render()).toMatchSnapshot();

  newTaskSubmit.simulate('submit');
  expect(wrapper.render()).toMatchSnapshot();

  const links = wrapper.find('[data-test="task-remove"]');
  links.last().simulate('click');
  expect(wrapper.render()).toMatchSnapshot();

  const link = wrapper.find('[data-test="task-toggle-state"]');
  link.simulate('click');
  expect(wrapper.render()).toMatchSnapshot();

  link.simulate('click');
  expect(wrapper.render()).toMatchSnapshot();
});
