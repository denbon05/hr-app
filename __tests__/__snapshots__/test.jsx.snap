// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store 1`] = `
<div
  class="col-5"
>
  <form
    class="form-inline"
  >
    <div
      class="form-group mx-3"
    >
      <input
        name="text"
        required=""
        type="text"
        value=""
      />
    </div>
    <input
      class="btn btn-primary btn-sm"
      type="submit"
      value="Add"
    />
  </form>
</div>
`;

exports[`Store 2`] = `
<div
  class="col-5"
>
  <form
    class="form-inline"
  >
    <div
      class="form-group mx-3"
    >
      <input
        name="text"
        required=""
        type="text"
        value="na-na"
      />
    </div>
    <input
      class="btn btn-primary btn-sm"
      type="submit"
      value="Add"
    />
  </form>
</div>
`;

exports[`Store 3`] = `
<div
  class="col-5"
>
  <form
    class="form-inline"
  >
    <div
      class="form-group mx-3"
    >
      <input
        name="text"
        required=""
        type="text"
        value=""
      />
    </div>
    <input
      class="btn btn-primary btn-sm"
      type="submit"
      value="Add"
    />
  </form>
  <div
    class="mt-3"
  >
    <ul
      class="list-group"
    >
      <li
        class="list-group-item d-flex"
      >
        <span
          class="mr-auto"
        >
          <button
            class="btn btn-link p-0 text-decoration-none"
            data-test="task-toggle-state"
            type="button"
          >
            na-na
          </button>
        </span>
        <button
          class="close"
          data-test="task-remove"
          type="button"
        >
          <span>
            ×
          </span>
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Store 4`] = `
<div
  class="col-5"
>
  <form
    class="form-inline"
  >
    <div
      class="form-group mx-3"
    >
      <input
        name="text"
        required=""
        type="text"
        value="another task"
      />
    </div>
    <input
      class="btn btn-primary btn-sm"
      type="submit"
      value="Add"
    />
  </form>
  <div
    class="mt-3"
  >
    <ul
      class="list-group"
    >
      <li
        class="list-group-item d-flex"
      >
        <span
          class="mr-auto"
        >
          <button
            class="btn btn-link p-0 text-decoration-none"
            data-test="task-toggle-state"
            type="button"
          >
            na-na
          </button>
        </span>
        <button
          class="close"
          data-test="task-remove"
          type="button"
        >
          <span>
            ×
          </span>
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Store 5`] = `
<div
  class="col-5"
>
  <form
    class="form-inline"
  >
    <div
      class="form-group mx-3"
    >
      <input
        name="text"
        required=""
        type="text"
        value=""
      />
    </div>
    <input
      class="btn btn-primary btn-sm"
      type="submit"
      value="Add"
    />
  </form>
  <div
    class="mt-3"
  >
    <ul
      class="list-group"
    >
      <li
        class="list-group-item d-flex"
      >
        <span
          class="mr-auto"
        >
          <button
            class="btn btn-link p-0 text-decoration-none"
            data-test="task-toggle-state"
            type="button"
          >
            another task
          </button>
        </span>
        <button
          class="close"
          data-test="task-remove"
          type="button"
        >
          <span>
            ×
          </span>
        </button>
      </li>
      <li
        class="list-group-item d-flex"
      >
        <span
          class="mr-auto"
        >
          <button
            class="btn btn-link p-0 text-decoration-none"
            data-test="task-toggle-state"
            type="button"
          >
            na-na
          </button>
        </span>
        <button
          class="close"
          data-test="task-remove"
          type="button"
        >
          <span>
            ×
          </span>
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Store 6`] = `
<div
  class="col-5"
>
  <form
    class="form-inline"
  >
    <div
      class="form-group mx-3"
    >
      <input
        name="text"
        required=""
        type="text"
        value=""
      />
    </div>
    <input
      class="btn btn-primary btn-sm"
      type="submit"
      value="Add"
    />
  </form>
  <div
    class="mt-3"
  >
    <ul
      class="list-group"
    >
      <li
        class="list-group-item d-flex"
      >
        <span
          class="mr-auto"
        >
          <button
            class="btn btn-link p-0 text-decoration-none"
            data-test="task-toggle-state"
            type="button"
          >
            another task
          </button>
        </span>
        <button
          class="close"
          data-test="task-remove"
          type="button"
        >
          <span>
            ×
          </span>
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Store 7`] = `
<div
  class="col-5"
>
  <form
    class="form-inline"
  >
    <div
      class="form-group mx-3"
    >
      <input
        name="text"
        required=""
        type="text"
        value=""
      />
    </div>
    <input
      class="btn btn-primary btn-sm"
      type="submit"
      value="Add"
    />
  </form>
  <div
    class="mt-3"
  >
    <ul
      class="list-group"
    >
      <li
        class="list-group-item d-flex"
      >
        <span
          class="mr-auto"
        >
          <button
            class="btn btn-link p-0 text-decoration-none"
            data-test="task-toggle-state"
            type="button"
          >
            <s>
              another task
            </s>
          </button>
        </span>
        <button
          class="close"
          data-test="task-remove"
          type="button"
        >
          <span>
            ×
          </span>
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Store 8`] = `
<div
  class="col-5"
>
  <form
    class="form-inline"
  >
    <div
      class="form-group mx-3"
    >
      <input
        name="text"
        required=""
        type="text"
        value=""
      />
    </div>
    <input
      class="btn btn-primary btn-sm"
      type="submit"
      value="Add"
    />
  </form>
  <div
    class="mt-3"
  >
    <ul
      class="list-group"
    >
      <li
        class="list-group-item d-flex"
      >
        <span
          class="mr-auto"
        >
          <button
            class="btn btn-link p-0 text-decoration-none"
            data-test="task-toggle-state"
            type="button"
          >
            another task
          </button>
        </span>
        <button
          class="close"
          data-test="task-remove"
          type="button"
        >
          <span>
            ×
          </span>
        </button>
      </li>
    </ul>
  </div>
</div>
`;
